### ISSUE TITLE
RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by saswatpatro** [ngx-formly/ngx-formly#2993]
Hi Team,
So i am getting an error - 
RenderFormComponent.html:5 ERROR RangeError: Maximum call stack size exceeded
    at RegExp.get global [as global] (<anonymous>)
    at RegExp.[Symbol.replace] (<anonymous>)
    at String.replace (<anonymous>)
    at FieldExpressionExtension.setExprValue (ngx-formly-core.js:2683)
    at FieldExpressionExtension.checkFieldExpressionChange (ngx-formly-core.js:2536)
    at ngx-formly-core.js:2499
    at Array.forEach (<anonymous>)
    at FieldExpressionExtension._checkField (ngx-formly-core.js:2491)
    at ngx-formly-core.js:2505
    at Array.forEach (<anonymous>)

in a form where I am using hooks for calling an API whenever there is a change in the value of a particular dropdown..The  form is having many dependent dropdowns each containing valueChanges in them which in turn helps me fetch data if a dropdown's value is changed.

I tried searching about it a lot but couldn't get a probable solution..If you could please help!
Thanks in advance!

**Comment by saswatpatro** [comment]
This is what the error looks like..
![error](https://user-images.githubusercontent.com/24622689/135209371-5c6d5df9-5164-43a4-bdda-6a99bb805956.JPG)

**Comment by saswatpatro** [comment]
Another scenario where this arises is when I am using expression properties for any calculation based on fields present in the forms..
Just a week back everything was working fine..Is it due to some latest releases (Just a doubt!)

**Comment by saswatpatro** [comment]
And yes I just tried doing the same thing with 5.10.22..It was working..But with 5.10.23 it is giving this error!
So may be the latest release is going into some infinite loop in checkFieldExpression?

**Comment by Letyr** [comment]
Got similar issue. I have expressionPropperties function, that writes another funtion to templateOptions like 
```
'templateOptions.someFunction': model => someBoolean ? () => { ...someFunctionBody } : undefined
```
as far as I can see the recursion start on this point https://github.com/ngx-formly/ngx-formly/commit/7209e0e#diff-5279aa0bed4f4b57ee43e615a4d08dd96c64d56bb0dee98e15a7d2d4f80a5703R141 , formly infinitely checks expressionProperties, because expressionProperties resolved value can't pass cache cheking here https://github.com/ngx-formly/ngx-formly/commit/7209e0e#diff-5279aa0bed4f4b57ee43e615a4d08dd96c64d56bb0dee98e15a7d2d4f80a5703L181  

![image](https://user-images.githubusercontent.com/23337339/135234817-8a36650c-15f2-4460-85f0-eb1ec8083908.png)

**Comment by saswatpatro** [comment]
Yes @Letyr this takes it into an infinite sort of a loop which exceeds the stack.
For now you can use version 5.10.22 which seems to be stable! :)

**Comment by aitboudad** [comment]
@saswatpatro could you please show us a snippet of code that produces that issue?

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.26](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.26) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
